keymap/vim.js

```diff
@@ -397,8 +397,9 @@
       }
     }
 
-    function setOption(name, value) {
+    function setOption(name, value, cm, cfg) {
       var option = options[name];
+      var scope = (cfg || {}).scope;
```

Comment:
not a fan of this structure: (a || {}).key
It seems to be used elsewhere, so I'll let it go, though.  Any changes here are optional.

My reasoning is that if you're scanning for places where the object is dereferenced, it's easy to miss.  Alternatives would be
cfg = cfg || {};
var scope = cfg.scope

or 

cfg = (cfg ? cfg.scope : null)